twisher/src/main/java/posts/ServletForPosts.java

```diff
@@ -0,0 +1,175 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+public class ServletForPosts extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Post post = Posts.posts1.getPostByID(id);
+        String str;
+        if(post == null)
+            str = "Error";
+        else
+            str = post.toString();
+        resp.getOutputStream().write(str.getBytes(StandardCharsets.UTF_8));
+
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        if(req.getRequestURL().indexOf("/add") != -1) {
+            if(addPost(req, resp))
+                resp.getOutputStream().println("Success");
+            else
+                resp.getOutputStream().println("Error");
+        }
+        else
+            postPage(req, resp);
+    }
+
+    private boolean addPost(HttpServletRequest req, HttpServletResponse resp) {
+        Post post = parsePost(req, resp);
+        if(post.getAuthor().equals(""))
+            return false;
+
+        post.setId(Posts.posts1.GetNext());
+        post.setCreatedAt(new Date());
+        return Posts.posts1.add(post);
+    }
+
+    private Post parsePost(HttpServletRequest req, HttpServletResponse resp) {
+        Post post = new Post();
+
+        String str = req.getParameter("au");
+        if(str != null)
+            post.setAuthor(str);
+        str = req.getParameter("d");
+        if(str != null)
+            post.setDescription(str);
+        str = req.getParameter("pl");
+        if(str != null)
+            post.setPhotoLink(str);
+        return post;
+    }
+
+    private void postPage(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        Integer skip, top;
+
+        String skipStr = req.getParameter("skip");
+        skip = parseInt(skipStr);
+        String topStr = req.getParameter("top");
+        top = parseInt(topStr);
+
+        Filter filter = new Filter();
+
+        String dateStr = req.getParameter("date");
+        if(dateStr != null) {
+            try {
+                filter.setCreatedAt(parseDate(dateStr));
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+        }
+        String author = req.getParameter("author");
+        if(author != null)
+            filter.setAuthor(author);
+
+
+
+        List<Post> posts = new ArrayList<Post>(Posts.posts1.getPosts(skip, top, filter));
+        ServletOutputStream out = resp.getOutputStream();
+        printTweets(out, posts);
+    }
+
+    private Integer parseInt(String str) {
+        if(str == null)
+            return null;
+        else
+            return Integer.parseInt(str);
```

Comment:
Still may be an exception if string contains other symbols than digits. Must wrap it in `try.. catch` of check for containing only digits